ckanext/security/authenticator.py

```diff
@@ -29,6 +29,31 @@ def get_login_throttle_key(request, user_name):
 
     return login_throttle_key
 
+
+def get_user_throttle(user_name):
+    if config.get('ckanext.security.brute_force_key') != 'user_name':
+        return {}
+    return LoginThrottle(User.by_name(user_name), user_name).get()
+
+
+def get_address_throttle(address):
+    if config.get('ckanext.security.brute_force_key') == 'user_name':
+        return {}
+    return LoginThrottle(None, address).get()
+
+
+def reset_user_throttle(user_name):
+    if config.get('ckanext.security.brute_force_key') != 'user_name':
```

Comment:
Couldn't this reuse the 'get' function?

Reply:
which 'get' function do you mean?

Reply:
The get_user_throttle function you're adding. That would avoid having to retrieve the same config key twice.

Reply:
Sorry @ThrawnCA I'm still not seeing it. Do you mind clicking the [±] button and showing me the change you mean?

Reply:
You're adding get_user_throttle at line 33. Couldn't it be used to simplify the logic at line 46?

Reply:
I don't think so. `get_user_throttle` returns the json stored in redis for the given user name, but `reset_user_throttle` clears that data. I can't clear the data stored in redis when all I have is a copy of the json data and not a LoginThrottle instance to call `.reset()`

I've left this code simple, repeating the configuration option checks close to each other in the same file in case we want to support throttling both user names and IP addresses at the same time in the future (we can simply delete these checks and namespace the user/address keys)

Reply:
Oh, I think I see, the get function isn't just returning the LoginThrottle object, it's calling 'get' on that first.